.github/workflows/build_ci.yml

```
@@ -274,6 +274,62 @@ jobs:
       shell: bash
       run: cargo publish -p dofigen --allow-dirty --token "${{ secrets.CARGO_TOKEN }}"
 
+  publish-homebrew:
+    name: Publish Homebrew
+    needs: [build, get-next-version, release]
+    if: github.ref_name == 'main'
+    runs-on: ubuntu-latest
+    env:
+      VERSION: ${{ needs.get-next-version.outputs.version }}
+      GITHUB_TOKEN: ${{ secrets.HOMEBREW_RELEASE_TOKEN }}
+    steps:
+    - name: Checkout
+      if: ${{ needs.get-next-version.outputs.will-release }}
+      uses: actions/checkout@v3
+
+    - name: download-artifact-macos-x86_64
+      if: ${{ needs.get-next-version.outputs.will-release }}
+      uses: actions/download-artifact@v4
+      with:
+        name: dofigen-macos-x86_64
+
+    - name: download-artifact-macos-aarch64
+      if: ${{ needs.get-next-version.outputs.will-release }}
+      uses: actions/download-artifact@v4
+      with:
+        name: dofigen-macos-aarch64
+
+    - name: download-artifact-linux-x86_64
+      if: ${{ needs.get-next-version.outputs.will-release }}
+      uses: actions/download-artifact@v4
+      with:
+        name: dofigen-linux-x86_64
+
+    - name: download-artifact-linux-aarch64
+      if: ${{ needs.get-next-version.outputs.will-release }}
+      uses: actions/download-artifact@v4
+      with:
+        name: dofigen-linux-aarch64
+
+    - name: Generate Homebrew file
+      if: ${{ needs.get-next-version.outputs.will-release }}
+      shell: bash
+      run: |
+        sed -i "s/\${VERSION}/${{ env.VERSION }}/g" .github/homebrew.rb
+        sed -i "s/\${MACOS_ARM_SHA256}/$(shasum -a 256 ./dofigen-macos-aarch64 | awk '{printf $1}')/g" .github/homebrew.rb
+        sed -i "s/\${MACOS_INTEL_SHA256}/$(shasum -a 256 ./dofigen-macos-x86_64 | awk '{printf $1}')/g" .github/homebrew.rb
+        sed -i "s/\${LINUX_ARM_SHA256}/$(shasum -a 256 ./dofigen-linux-aarch64 | awk '{printf $1}')/g" .github/homebrew.rb
+        sed -i "s/\${LINUX_INTEL_SHA256}/$(shasum -a 256 ./dofigen-linux-x86_64 | awk '{printf $1}')/g" .github/homebrew.rb
+        cat .github/homebrew.rb
```

It's Ok for me, but you can change this with :

```suggestion
    - name: download-artifacts
      if: ${{ needs.get-next-version.outputs.will-release }}
      uses: actions/download-artifact@v4
      with:
        path: artifacts
        pattern: dofigen-[linux|macos]-[aarch64|x86_64]
        merge-multiple: true
        
    - name: Generate Homebrew file
      if: ${{ needs.get-next-version.outputs.will-release }}
      shell: bash
      run: |
        sed -i "s/\${VERSION}/${{ env.VERSION }}/g" .github/homebrew.rb
        sed -i "s/\${MACOS_ARM_SHA256}/$(shasum -a 256 ./artifacts/dofigen-macos-aarch64 | awk '{printf $1}')/g" .github/homebrew.rb
        sed -i "s/\${MACOS_INTEL_SHA256}/$(shasum -a 256 ./artifacts/dofigen-macos-x86_64 | awk '{printf $1}')/g" .github/homebrew.rb
        sed -i "s/\${LINUX_ARM_SHA256}/$(shasum -a 256 ./artifacts/dofigen-linux-aarch64 | awk '{printf $1}')/g" .github/homebrew.rb
        sed -i "s/\${LINUX_INTEL_SHA256}/$(shasum -a 256 ./artifacts/dofigen-linux-x86_64 | awk '{printf $1}')/g" .github/homebrew.rb
```

I removed the latest `cat` command because I think you forget it ?

---

Actually the cat was for debugging but it's true that it's not really necessary.

We need to make sure that the patterns works. I'll make some tests